src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs

```diff
@@ -25,15 +25,15 @@ public LinuxAppServiceEventGenerator(LinuxAppServiceFileLoggerFactory loggerFact
 
         public override void LogFunctionTraceEvent(LogLevel level, string subscriptionId, string appName, string functionName, string eventName,
             string source, string details, string summary, string exceptionType, string exceptionMessage,
-            string functionInvocationId, string hostInstanceId, string activityId, string runtimeSiteName, string slotName)
+            string functionInvocationId, string hostInstanceId, string activityId, string runtimeSiteName, string slotName, DateTime eventTimestamp)
         {
-            var eventTimestamp = DateTime.UtcNow.ToString(EventTimestampFormat);
+            var formattedEventTimestamp = eventTimestamp.ToString(EventTimestampFormat);
```

Comment:
Because I'm now flushing logs in the background, so the actual event timestamp is now captured when the log is written. It'll be flushed later.

Reply:
I undid the batch flushing, but left the structural change of taking the event timestamp as a param since that's a good change anyways - all event data logged coming from caller.